### PR TITLE
Support for Beta Language Banner

### PR DESCRIPTION
**Pull request by nasthagiri**
https://openedx.atlassian.net/browse/ARCH-716

Screen Shot 2019-05-10 at 5 01 09 PM

Note: This may need an additional pass to improve its mobile responsiveness.

### CODE REVIEW

File: src/account-settings/AccountSettingsPage.jsx [hunk 1]
```diff
@@ -317,7 +378,12 @@ AccountSettingsPage.propTypes = {
     social_link_twitter: PropTypes.string,
     time_zone: PropTypes.string,
   }).isRequired,
-  siteLanguage: PropTypes.string,
+  userLocale: PropTypes.string,
+  siteLanguages: PropTypes.arrayOf(PropTypes.shape({
+    name: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```
Comment (robrap) [inline src/account-settings/AccountSettingsPage.jsx]: Just curious why these are string or number?

File: src/account-settings/selectors.js [hunk 2]
```diff
@@ -136,40 +141,50 @@ const countryTimeZonesSelector = createSelector(
...
+  previousLocaleSelector,
   draftsSelector,
   localeSelector,
-  (drafts, locale) => (drafts.siteLanguage !== undefined ?
-    drafts.siteLanguage : getAssumedServerLanguageCode(locale)),
+  (previousValue, drafts, locale) => {
+    const savedValue = getAssumedServerLanguageCode(locale);
+    return {
+      previousValue,
+      draftValue: (drafts.userLocale !== undefined ? drafts.userLocale : savedValue),
```
Comment (robrap) [inline src/account-settings/selectors.js]: Curious if `drafts.userLocale || savedValue` works?

File: src/site-language/selectors.js [hunk 3]
```diff
@@ -1,17 +1,17 @@
 import { createSelector } from 'reselect';
 
-export const storeName = 'siteLanguage';
+export const storeName = 'siteLanguages';
 
-const siteLanguageSelector = state => state[storeName];
+const siteLanguageStoreSelector = state => state[storeName];
```
Comment (robrap) [inline src/site-language/selectors.js]: siteLanguage**s**StoreSelector? (i.e. languages plural?)

File: src/account-settings/AccountSettingsPage.jsx [hunk 4]
```diff
@@ -64,10 +65,25 @@ class AccountSettingsPage extends React.Component {
     return concatTimeZoneOptions;
   });
 
+  getTransifexLink(languageCode) {
+    const tokenizedCode = languageCode.split('-');
+    let languageSubPage = tokenizedCode[0];
+    if (tokenizedCode.length > 1) {
+      languageSubPage += `_${tokenizedCode[1].toUpperCase()}`;
+    }
+    return TRANSIFEX_LANGUAGE_BASE_URL + languageSubPage;
```
Comment (robrap) [inline src/account-settings/AccountSettingsPage.jsx]: Through test or comment, could you add an example of languageCode => url, to better understand what this is doing?

File: src/account-settings/AccountSettingsPage.jsx [hunk 5]
```diff
@@ -115,10 +116,10 @@ class AccountSettingsPage extends React.Component {
 
     return (
       <div>
+        {this.renderBanner()}
```
Comment (abutterworth) [inline src/account-settings/AccountSettingsPage.jsx]: I think you can swap this now with:
```suggestion
        <BetaLanguageBanner />
```

File: src/account-settings/AccountSettingsPage.jsx [hunk 6]
```diff
@@ -230,7 +231,7 @@ class AccountSettingsPage extends React.Component {
               name="siteLanguage"
               type="select"
               options={this.props.siteLanguageOptions}
-              value={this.props.siteLanguage}
```
Comment (abutterworth) [inline src/account-settings/AccountSettingsPage.jsx]: Naming-wise, Maybe `this.props.siteLanguage.value` would be more similar to the other fields? From here: https://github.com/edx/frontend-app-account/pull/24/files#diff-e7032fb7d22a11399f664effe094e9beR152

Comment (nasthagiri) [reply src/account-settings/AccountSettingsPage.jsx]: Per our conversation, I'll change this to `draftOrSavedValue` - in a follow-up PR.

**Pull request merged**